commitment/taproot.go

```diff
@@ -73,6 +74,7 @@ func (t TapscriptPreimageType) String() string {
 // TapscriptPreimage wraps a pre-image byte slice with a type byte that self
 // identifies what type of pre-image it is.
 type TapscriptPreimage struct {
+	// TODO(jhb): SiblingPreimage need not be 32 bytes?
```

Comment:
So the sibling preimage here assumes a `tapHash` round has already been done when it's a leaf. So if it was actually a leaf, then it'll be 32 bytes. If two branches, then 64 bytes.

Reply:
We could also try to unroll/enumerate all the possible versions here in diff types instead of the enum as well. I think we can do that in another pass though, as then we can tackle decomposing some of the optional/inner structs in the proof object.

Reply:
Talked to Oli, and the comment here is wrong - if the preimage is created from a leaf, it must be the full leaf. Rather, the preimage cannot be a taphash, because then we  wouldn't be able to verify that it's not a tap asset commitment.

This was the issue I hit before when modifying the proof verification code, where i'd already taphashed the preimage, but the verifier needs to do that hashing itself.